tools/mysql-tests/rds/iam_auth.go

```diff
@@ -0,0 +1,86 @@
+//nolint:gocritic // Test tool, not production code
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/datastore/mysql/common_mysql"
+	kitlog "github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/jmoiron/sqlx"
+)
+
+var (
+	endpointFlag   = flag.String("endpoint", "", "RDS endpoint address (without port)")
+	portFlag       = flag.String("port", "3306", "Database port")
+	userFlag       = flag.String("user", "fleet_iam_user", "Username for IAM authentication")
+	dbNameFlag     = flag.String("db", "fleet", "Database name")
+	assumeRoleFlag = flag.String("assume-role", "", "STS assume role ARN (optional)")
+	externalIDFlag = flag.String("external-id", "", "STS external ID (optional)")
+)
+
+func main() {
+	flag.Parse()
+
+	if *endpointFlag == "" {
+		log.Fatal("RDS endpoint is required (-endpoint flag)")
+	}
+	if *userFlag == "" {
+		log.Fatal("Username is required (-user flag)")
+	}
+
+	logger := level.NewFilter(kitlog.NewLogfmtLogger(os.Stderr), level.AllowDebug())
+
+	// Configure MySQL connection with IAM auth
+	mysqlConfig := &config.MysqlConfig{
+		Protocol:         "tcp",
+		Address:          fmt.Sprintf("%s:%s", *endpointFlag, *portFlag),
+		Username:         *userFlag,
+		Database:         *dbNameFlag,
+		StsAssumeRoleArn: *assumeRoleFlag,
+		StsExternalID:    *externalIDFlag,
+	}
```

Comment:
Looks like this tool was not updated for the change to require specifying the region. It should either be updated or removed if you don't plan on using it for QA. I included it because I used it during development.